packages/code-studio/src/styleguide/StyleGuide.tsx

```diff
@@ -48,25 +48,28 @@ const stickyProps = {
 } as const;
 
 function StyleGuide(): React.ReactElement {
-  const isolateSection = window.location.search.includes('isolateSection=true');
   const { themes } = useTheme();
+  const [hash, setHash] = useState(window.location.hash);
   const hasMultipleThemes = themes.length > 1;
 
+  function isHash(label: string, processLabel = false): boolean {
```

Comment:
Instead of passing this around via props, I would consider encapsulating the hash matching functionality in a custom hook. Something like:

```typescript
function useIsHash() {
  const [hash, setHash] = useState(window.location.hash);

  useEffect(() => {
    const hashChangeHandler = () => setHash(window.location.hash);
    window.addEventListener('hashchange', hashChangeHandler);
    return () => window.removeEventListener('hashchange', hashChangeHandler);
  }, []);

  const isHash = useCallback(
    (label: string, processLabel = false): boolean => {
      const newLabel = processLabel
        ? label.toLocaleLowerCase().replaceAll(' ', '-')
        : label;

      return hash === `#${newLabel}` || hash === '';
    },
    [hash]
  );

  return isHash;
}
```

Reply:
Then you can just use it in any component like:
```
const isHash = useIsHash();
```

Reply:
is there something that can be done so that it doesn't have to be littered all over the place?